### ISSUE TITLE
Development Discussion

### ISSUE THREAD

**Issue by lexruee** [LinxGem33/Arc-Menu#32]
@LinxGem33 
Currenlty, the master branch is broken. In other words, the last commit introduced two syntax errors. I've added some comments: https://github.com/LinxGem33/Arc-Menu/commit/748a6395d53360cf362d5bbb9dcaf7ee5a2e7d09

I think this is a good occasion  to create a `development` branch or short a `dev` branch on your upstream repository. So if something breaks on the dev branch it is not so bad. At the same, we ensure that the master branch is always working.

**Comment by LinxGem33** [comment]
@lexruee 

appreciate the feed back Ive amended the changes and created a [development branch](https://github.com/LinxGem33/Arc-Menu/tree/Development-Branch) let me know if this is ok 

kind regards

**Comment by lexruee** [comment]
@LinxGem33 
Holy moly, this was fast response ;-). That's fine.

**Comment by LinxGem33** [comment]
ha no problem , :relaxed:

**Comment by lexruee** [comment]
@LinxGem33 

**Testing**
If you need some extra testers for future releases on extensions.gnome.org let me know. 
On my development machine, I have setup some Debian/Ubuntu virtual machines. However, I'm thinking about how we could run automated tests using docker.

**Binding the Super key to the Arc Menu**
At the moment, I have a very _limited_ working solution for binding the Super key to the menu: https://github.com/LinxGem33/Arc-Menu/issues/7
In other words, you can hit the Super Key and the menu pops up. Better than nothing I guess :-).
I'll make soon a new branch on my github repo and share what I have so far.

**Comment by LinxGem33** [comment]
@lexruee 
##
**RE**: Binding the Super key to the Arc Menu
maybe a solution like the one gnomenu could work for us as it has the code already there to add to the extension and also the prefs.js file and it would save us a lot of time in trying to figure out the correct API's etc, also there is a lot of code that could be integrated into Arc menu after further examining it, what do you think?
##
**RE**: Testing
any help is welcome. :smile: 
##
**Final thoughts**!
What are your thoughts on taking some of the code from gnomenu?
##
kind regards

**Comment by lexruee** [comment]
@LinxGem33 

Good news! I managed to implement the Super Key feature for the Arc Menu. Here is a sneak preview :-).
![screenshot from 2017-06-26 09-53-30](https://user-images.githubusercontent.com/649340/27529606-7065bffe-5a55-11e7-9d82-86dee4f0a385.png)

![screenshot from 2017-06-26 09-53-48](https://user-images.githubusercontent.com/649340/27529618-80c39aa6-5a55-11e7-9d97-d69e0afe8d39.png)
Before I make a pull-request, I have to refactor some things and test it some virtual machines. 
Overall, I think this feature is fully implemented.

**Comment by LinxGem33** [comment]
@lexruee 
##
congratulations, fantastic news, you have done a great job I’m looking forward to merging your PR Im working on hacking a bit more code from the gnomenu for you it should help with a few more features as I’m currently still a bit confused on how to compile the changes into gsettings etc but in the meantime ill post the code here for you.
##
**Ideas**
a couple of ideas regarding the suspend button, i was thinking to add a disable button on the prefs.js for the suspend button as we could reuse the code from the (disable hot corner code) and modify it for the suspend button.
##
kind regards

**Comment by lexruee** [comment]
**RE: Testing**
I'm almost done with testing the latest Development-Branch with the merged changes on my repository:
https://github.com/lexruee/Arc-Menu/tree/Development-Branch


|OS                       | gnome-shell  | branch               | works     |  remakrs
|-------------------------|--------------|----------------------|-----------|-------
| Debian Jessie           | 3.14         | Development-Branch   | no        | Error: Shell.ActionMode is undefined
| Debian Stretch          | 3.22         | Development-Branch   | yes       |
| Debian Sid (my system)  | 3.22         | Development-Branch   | yes       |
| Ubuntu Gnome LTS 16.04  | 3.18         | Development-Branch   | yes       |
| Ubuntu Gnome 17.04      | 3.24         | Development-Branch   | yes       |
| ~~Arch Linux~~              | ~~3.24~~          | ~~Development-Branch~~    | TBA    | Not tested yet.


**Bug: Debian Jessie/gnome-shell 3.14**
It appears that Arc Menu does not work on Debian Jessie with gnome-shell 3.14. Looking glass shows the error: Shell.ActionMode is undefined. I guess this can be easily fixed with a small compatibility hack.

![debian-jessie-bug](https://user-images.githubusercontent.com/649340/27562281-261c152a-5acc-11e7-97fd-0b8998a04c53.png)


**Bug: Activities Label**
During my testing procedure, I encountered a small bug. If Arc Menu is deactivated, then the Activities Label is not at the default position.

Before Arc-Menu installation:
![before_installation](https://user-images.githubusercontent.com/649340/27561642-3557f638-5ac9-11e7-9d58-096c6fe85d88.png)

Installed:
![installed](https://user-images.githubusercontent.com/649340/27561648-3943e7c0-5ac9-11e7-94ab-26932688ff4f.png)

Deactivated:
![decativated](https://user-images.githubusercontent.com/649340/27561653-3e17a6ba-5ac9-11e7-9a4c-028f6069e63c.png)

**RE: Ideas**
An enable/disable for the suspend button should be simple to implement. As you said, we can now reuse the existing code as a skeleton :-). I must say, I really enjoyed working on your extension and reading some gnome-shell sourcecode.

**RE: Final thoughts**
I looked at the Gnomenu code. To be honest, I didn't really like reading the Gnomenu sourcecode. For learning the basic stuff, I looked at the source code of Dash-to-Dock, Workspaces-to-Dock, and some simple ones like Caffeine. In my opinion, Dash-to-Dock and Workspaces-to-Dock are well-written extensions.

**Comment by LinxGem33** [comment]
@lexruee 
##
brilliant I’m going to test now on my setup, you have done a really good job so fa and i really appreciate the time and effort so far, ill see if i can create some code for a couple other features to include.
## 
kind regards

**Comment by LinxGem33** [comment]
@lexruee 
##
there is just one issue you need to compile the schema as it wont load on the gnome shell otherwise please see below the error message.
##
![error](https://user-images.githubusercontent.com/19158615/27563982-98f547ec-5acc-11e7-859d-9aecac16ac08.png)
##
kind regards

**Comment by lexruee** [comment]
@LinxGem33 

You can use the Makefile to compile the schema:
```
make compile
```

When you are testing you can also use the install/uninstall commands:

```
make uninstall
make install
```

**Comment by LinxGem33** [comment]
apologies :blush:

**Comment by LinxGem33** [comment]
@lexruee 
##
its now working perfect :smile: 
##
![dgs](https://user-images.githubusercontent.com/19158615/27564477-d8a80aca-5acf-11e7-93af-db5c52628f74.png)
##
kind regards

**Comment by LinxGem33** [comment]
@lexruee 
##
do you feel comfortable enough to release this upstream to extensions.gnome.org the reason i ask is a code review is due in the next day or so i think so it will save us waiting another week or so for another one.
##
what do you think ?
##
kind regards

**Comment by lexruee** [comment]
@LinxGem33 

Never underestimate the power of `make`. It is old but does the job :sunglasses:. The prefs.js looks nice with your theme. 

I'll try to fix the Debian Jessie bug since the metadata.json file says that Arc Menu supports gnome-shell 3.14. Otherwise we should drop the support for gnome-shell 3.14.

I would feel more comfortable if some users in https://github.com/LinxGem33/Arc-Menu/issues/7 are willing to test it before it is released to extensions.gnome.org. It is up to you.You must know,  I'm risk averse. However, if you are okay with bug reports and frustrated users, then go ahead :smile:

**Comment by LinxGem33** [comment]
@lexruee 
##
no problem we can upload and see what happens the reason is some users wont know how to manually install and if i upload upstream to extensions.gnome.org i can always deactivate and revert back to the last known stable version so there is no harm done.

i agree we should drop support for 3.14 and maybe 3.16 as the support will end soon anyway i also upload some more code here for some of the features your working on maybe it will be useful.
##
kind regards

**Comment by LinxGem33** [comment]
@lexruee 

latest changes now merged upstream to extensions.gnome.org, everything is working on all my virtual testing machine its looking solid.

kind regards

**Comment by lexruee** [comment]
@LinxGem33 

That is great! Finally, my mum can have a useful application menu on her Gnome 3 Desktop. This is my main motivation why I contribute to this project :smile:. When it comes to translating the extension to German/French, I can help as well.

**Comment by LinxGem33** [comment]
@lexruee 

its definitely a good reason the better we make the menu the more people will find using gnome 3 easier who are used to that sort of applications menu, regarding the translations that would be a great help thank you.

**Comment by LinxGem33** [comment]
@lexruee 
##
**Code for Arc Menu's Position on the Panel** ( Left ,Right ,Centre )

I’ve put together this code hopefully it should help you to integrate the required code to set the position of Arc menu in the panel.
##
schema - code for the position
##
```xml
<?xml version="1.0" encoding="UTF-8"?>
<schemalist gettext-domain="gnome-shell-extensions-arc-menu">
  <enum id='org.gnome.shell.extensions.arc-menu.visible'>
    <value value="0" nick="left"/>
    <value value="1" nick="center"/>
    <value value="2" nick="right"/>
  </enum>
    </key>
    <key name="position-in-panel" enum="org.gnome.shell.extensions.arc-menu.position">
      <default>'left'</default>
      <summary>Position in Panel</summary>
    </key>
  </schema>
</schemalist>
```
##
prefs.js - code to set the position of Arc menu on the panel either ,left,right or centre 
##
```javascript
let presentLabel = '<b>' + _("Display Options") + '</b>';
        this.add(new Gtk.Label({
            label: presentLabel, use_markup: true,
            halign: Gtk.Align.START
    }));

    vbox.add(checkContainer);

    let positionContainer = new Gtk.HBox({spacing: 5});
    let positionLabel = new Gtk.Label({label: _('Arc Menu Position on the Panel')});
    let positionSelector = new Gtk.ComboBoxText();

    positionContainer.pack_start(positionLabel, 0,0,0);
    positionContainer.pack_end(positionSelector, 0,0,0);

    ['left','center','right'].forEach(function(item) {
      positionSelector.append_text(item);
    });

    positionSelector.set_active(this._settings.get_enum(SETTINGS_POSITION));

    let self = this;

    positionSelector.connect('changed', function(pos) {
      self._settings.set_enum(SETTINGS_POSITION, positionSelector.get_active());
    });
```
##
menu.js - code for the position
##
```javascript
const SETTINGS_POSITION = 'position-in-panel';

  resetPanelPos: function() {

    Main.panel.statusArea['Arc menu'] = null;
    Main.panel.addToStatusArea('Arc menu', this, 1, this._menuPosition);

  },

  setPrefs: function() {
 
   this._prevMenuPosition = this._menuPosition;

    this._menuPosition = this._settings.get_string(SETTINGS_POSITION);
  },
```
##
kind regards

**Comment by LinxGem33** [comment]
@lexruee 

nearly finished the home page for the website how are you doing have you managed to make any progress?

kind regards

**Comment by lexruee** [comment]
@LinxGem33 

**Merging development branch into master branch**
What do think about merging the development branch into the master branch?
Just have a look at this interesting branching tree: https://github.com/LinxGem33/Arc-Menu/network
And of course we should tag the updatet master branch with the latest release number 13 from https://extensions.gnome.org/review/7057 . 13 is a special release number :smile:.

**Current state of the code base**
Overall, I think we should have a solid code base, which should help us to implement new features. As you might noticed I like to refactor and improve code :smile:.  Ideally, well written code is self-documenting, at least in theory.

In contrast to the other projects (i.e., dash-to-dock, dash-to-panel) , the GUI of our preferences widget is purely handcrafted :smile:. For example, dash-to-dock has a GUI (Settings.ui file) which was created with Anjuta IDE or Glade.

**Comment by LinxGem33** [comment]
@lexruee 
##
**RE: interesting branching tree**:

lol ha its a work of art that’s the next thing to push to github for review. ill merge the changes and follow the naming convention for version etc.
##
**RE: Current state of the code base**

i agree it should make things easier to implement in future development cycles and it helps to follow along with any changes made to the code.
##
also I’ve updated the main README page check it out and let me know.

**Comment by lexruee** [comment]
@LinxGem33 

> I'm pleased to announce a new member of the team (lexruee (Xander) and now a lead developer in the design and development of this extension

Hey, that is cool :smile: but it also sounds like a lot of responsibility, especially since gnome-shell extensions can easily break gnome-shell compatibility. I guess with great power comes great responsibility (:spider: + :man:).

Anyways, Arc Menu is and will be a kick-ass gnome-shell extension. Recently, I was thinking if it is possible to implement a cortona-like feature using the [Alexa Voice Service](https://developer.amazon.com/alexa-voice-service). Not that I need this feature, I would like to know whether it can be done with [gjs](https://wiki.gnome.org/action/show/Projects/Gjs?action=show&redirect=Gjs).

**Comment by LinxGem33** [comment]
@lexruee 
##
>  I guess with great power comes great responsibility

there is no pressure on you to deliver just a big thank you for your work so far :smile: 
##
*RE: Alexa Voice Service*

as far as i know Gjs is based on spider monkey which is Firefox’s API for JavaScript which is then translated for Gtk and gnome-shell development  through Gjs to implement this feature would be a headline act in the Linux world but also very hard to translate Alexa's API's to gnome as there maybe some code which we just don’t have an equivalent in gnome development library which means it would have to be custom built.

**Comment by lexruee** [comment]
@LinxGem33 

**Mycroft**
Thanks for the link of the Mycroft project. That is really an interesting open source project. At home, I have several several Raspberry Pi's (made in UK ;-)) because of a home sensor network project. So experimenting with the Mycroft Artificial Intelligence platform on Raspberry Pi is definitely a future weekend project :smile:. In other words, realizing an own Voice Service API using Mycroft is a new item on my DIY project list.

**Gnome Shell architecture**
>as far as i know Gjs is based on spider monkey which is Firefox’s API for JavaScript which is then translated for Gtk and gnome-shell development through Gjs

That is right. Gjs is based on spider monkey. The Gnome project uses GObject introspection, which is a middleware layer between C libraries and language bindings, such as [Python-Gtk](https://python-gtk-3-tutorial.readthedocs.io/en/latest/) or [JavaScript-Gtk](http://gjs-tutorial.readthedocs.io/en/latest/index.html).

If you are interested in more details, there is a great blog post about the Gnome Shell architecture: 
http://blog.fpmurphy.com/2011/05/more-gnome-shell-customization.html#sthash.OgG3JkyB.W2DbtbBv.dpbs
This blog post helped me greatly to have a better understanding of the whole Gnome Shell ecosystem.
![](http://blog.fpmurphy.com/blog-images/gnome-shell-diagram.png)

**RE: RE: Alexa Voice Service**
***Little feasibility study***
I did only little research but theoretically it should be possible to implement a JavaScript/Gjs wrapper around the Alexa Voice Service. Following this [Alexa tutorial here](https://miguelmota.com/blog/alexa-voice-service-with-curl/), all we need is basically a JavaScript HTTP request library.

Fortunately, there is the [libsoup project](https://wiki.gnome.org/action/show/Projects/libsoup?action=show&redirect=LibSoup). In short, libsoup is an HTTP client/server library for GNOME. And the interesting and cool thing is, there is also a JavaScript language binding for libsoup: https://people.gnome.org/~gcampagna/docs/Soup-2.4/index.html

What I don't know so far is how we can record an audio signal using a microphone. But I assume that this can be solved using the [GStreamer](https://wiki.gnome.org/Projects/GStreamer) media processing framework.

***Licensing and legal issues***
> and there is also the issue with licensing some parts maybe open source but others maybe restricted with a closed source licence but again I’m not sure how there licence is structured this would be things to find out if we did decide to add this feature.

You are right! I'm sorry, I was naive about this alexa-feature. I'm' not a lawyer but I'm 100% sure there would be licensing and legal issues if we add this feature to Arc-menu. Long story short, for using the Alexa Voice Service, the project and all users have to agree to the [Alexa Voice Service Agreement](https://developer.amazon.com/public/solutions/alexa/alexa-voice-service/support/terms-and-agreements). 

As you wrote, we could work around these licensing and legal issues by separating the problematic code that uses the Alexa Voice Service in a second unofficial github project. In this case, this alexa-feature would not be part of the official Arc-Menu project but rather an addon to Arc-Menu. However, I feel this kind of defeats the whole idea of implementing the alexa-feature... Thus, your mentioned Mycroft approach is of course the better way to go if we want to add this feature.

Lesson learned: I see writing free software is not an easy task when non-free Web Services are involved:
https://www.gnu.org/philosophy/network-services-arent-free-or-nonfree.html

**Comment by LinxGem33** [comment]
@lexruee 
##
yeah mycroft its definitely a good choice for all sorts situations and the best part its free and open source, thanks for sharing the links to the gnome shell architecture blog post, i was thinking that we really haven’t got that much left to finish with the menu its cool that just a few months ago this menu didn’t even exist "that’s why i love open source" side note I’ve also signed up to a pure JavaScript & ES6 & ES7 course to brush up on my coding skills so i can hopefully start making some more cool extensions.
##
Re: the remaining features 

how are you coming on with the AZ feature i have left you some code to review #50  its just a start on the prefs.js element of it but it might help let me know what you think.

**Comment by lestcape** [comment]
Hi. See, if we replace the activity button with the arc-menu, without check if activity button is present or not, the arc-menu can not be loaded if is not present...

Proposition: Check if Activity button is present and if not add the arc-menu to the first position in the panel.

**Comment by lexruee** [comment]
@lestcape 
Can you provide more details how you were able to produce this error? And maybe create a issue in this repository?

The code in controller.js is responsible for disabling and enabling the Arc Menu button as well as the Activities button: 
* see https://github.com/LinxGem33/Arc-Menu/blob/master/controller.js#L44
* see https://github.com/LinxGem33/Arc-Menu/blob/master/controller.js#L230
* see https://github.com/LinxGem33/Arc-Menu/blob/master/controller.js#L260

Currently, there are only simple checks to test whether the Activities button is present on the main panel based on the `_activitiesButtonIsRemoved` boolean flag.

But you are right, the current solution is not fail-proof in case another extension misbehaves or enables/disables the Activities button.

A fail-proof solution should test if the Activities button  is contained in `Main.panel._leftBox` (see gnome-shell [panel.js#L784](https://github.com/GNOME/gnome-shell/blob/master/js/ui/panel.js#L784 )). This `_leftBox` object is of type [St.BoxLayout](https://www.roojs.com/seed/gir-1.2-gtk-3.0/seed/St.BoxLayout.html) and has a `contains` method, which is inherited from [ClutterActor#contains method](https://developer.gnome.org/clutter/stable/ClutterActor.html#clutter-actor-contains).

Long story short, I would adapt the code in controller.js as follows to make it more robust:
```javascript
...
    // Chek if the activities button is present on the main panel    
    _isActivitiesButtonPresent: function () {
             return Main.panel._leftBox.contains(this._activitiesButton.container);
    },

    // Remove the activities button from the main panel
    _removeActivitiesButtonFromMainPanel: function() {
        if (this._isActivitiesButtonPresent()) {
            Main.panel._leftBox.remove_child(this._activitiesButton.container);
        }
    },

    // Add or restore the activities button on the main panel
    _addActivitiesButtonToMainPanel: function() {
        if (!this._isActivitiesButtonPresent()) {
            // Retsore the activities button at the default position
            Main.panel._leftBox.add_child(this._activitiesButton.container);
            Main.panel._leftBox.set_child_at_index(this._activitiesButton.container, 0);
        }
    },
...
```
Note: The code above is not tested :smile:!

**Comment by lestcape** [comment]
@lexruee what occurs to you man? You are pretty professional, that is not normal. Congratulation... 

Well i'm working porting cinnamon to gnome shell and my code remove the activity button and i see the problem...

I think you need to test if exist this._activitiesButton... that was the error.

**Comment by lestcape** [comment]
Something like that:
```
// Check if the activities button is present on the main panel    
    _isActivitiesButtonPresent: function () {
             return this._activitiesButton  &&  this._activitiesButton.container && Main.panel._leftBox.contains(this._activitiesButton.container);
    },

    // Remove the activities button from the main panel
    _removeActivitiesButtonFromMainPanel: function() {
        if (this._isActivitiesButtonPresent()) {
            Main.panel._leftBox.remove_child(this._activitiesButton.container);
        }
    },

    // Add or restore the activities button on the main panel
    _addActivitiesButtonToMainPanel: function() {
        if (!this._isActivitiesButtonPresent()) {
            // Retsore the activities button at the default position
            Main.panel._leftBox.add_child(this._activitiesButton.container);
            Main.panel._leftBox.set_child_at_index(this._activitiesButton.container, 0);
        }
    },
```

**Comment by lexruee** [comment]
@lestcape 
I like your solution :+1:. Very concise :smile:! I would suggest you to make a pull-request to the development branch.

In the future, I think the gnome people should provide the extension developers a better API to the Main.panel so we don't need to touch the internals of it. 

**_Offtopic:_**
You know, as a software developer I like to say that programming is a craft and creative problem solving :smile:. So, thanks for your compliment.

- I'm really interested in your [Gnome-Global-AppMenu](https://github.com/lestcape/Gnome-Global-AppMenu) project. I have not tested it but a Unity-like menu on GNOME shell is pretty cool.

- Regarding your Cinnamon-Porting-Project, can you provide me a link to a github repo. Maybe I can help you out on some things.

**Comment by lestcape** [comment]
Well when you create a pull request, add this please... If i will add something more, i will added this also, but just for one thing not.

This is the project: https://github.com/lestcape/Classic-Gnome. The images is real, but not all is unrelated to cinnamon yet. So, you need to install cinnamon, because it need some dependencies. Gsettings dependencies.

You explain all to the level of details. I can not do that. I considered it's an amassing quality that i don't have.

**Comment by lexruee** [comment]
**@LinxGem33** 


**Super Key binding feature**
Regarding the Super Key binding feature, we need to do some refinement. To solve the issues mentioned in https://github.com/LinxGem33/Arc-Menu/issues/7, we need, unfortunately, a solution that intervenes more deeply in the GNOME Shell :cry:.

**Eye-Candy-Feature: Transparency option**
Personally, I'm a big fan of the Tilix terminal. For example, Tilix allows the user to change the 
transparency of the background color. It would be cool if we could have the same feature in Arc Menu.

The "gimped" screenshot below shows the Tilix terminal in action and Arc Menu with some fake transparency:
![example_transparency](https://user-images.githubusercontent.com/649340/28066591-e681a6d2-663d-11e7-8e7f-5de240c17fdd.png)

Here is a mockup of the appearance page, which I made with [pencil](https://github.com/evolus/pencil):
![prefs-appearance](https://user-images.githubusercontent.com/649340/28065915-0732cbc0-663b-11e7-8dd9-5feb0d452227.png)

**Comment by LinxGem33** [comment]
@lexruee 
##
> Super Key binding feature
Regarding the Super Key binding feature, we need to do some refinement. To solve the issues mentioned in #7, we need, unfortunately, a solution that intervenes more deeply in the GNOME Shell :cry:.

just let me know how i can help, also i will start documenting the development of this extension to maybe help other new developers have a point of reference that may help them i will do little bit's at a time.
##
#### Re: Proposal Eye-Candy-Feature: Arc Menu transparency option

i think this is a fantastic idea :+1: 
##
#### Possible solution for feature (4) on the [feature list](https://github.com/LinxGem33/Arc-Menu/issues/42)

there is a application called Menulibre which allows the user to manipulate the applications menu and it works well with Arc menu i was thinking that we could add a install button on the prefs.js (in the fine tune page section) that triggers a install script to install  Menulibre from the official repositories and once installed there would be  a configure button that opens the program (i.e it just a button link to the program) then the user can configure and fine tune the menu i will do a wire frame prototype mock-up to illustrate my point :smile: 

let me know what you think of this idea hopefully ill have a mock up soon :crossed_fingers: 
##
**Menulibre Image below**

![mlib](https://user-images.githubusercontent.com/19158615/28067816-c03f973e-663a-11e7-9be3-bcac3405e747.png)

**Comment by LinxGem33** [comment]
@lexruee 

also what do you think about setting up a chat client (do you know any good one's) so we can talk to each other in real time i think it might help when developing features etc.

**Comment by lexruee** [comment]
@LinxGem33 
That is a good idea. I heard that MSN/ICQ is still the prefered chat client. Just kidding  :smile:.
I think a cheap solution would be IRC or something Web-based like slack.

**Comment by LinxGem33** [comment]
@lexruee 

(:laughing:  MSN ) ill set up IRC and send you details via (emal).

**Comment by LinxGem33** [comment]
@lexruee 

i have chosen Wire as the chat client its open source and secure etc - Link to download -[Wire](https://wire.com/en/download/)

let me know when you have set-up and send a invite. :smile:

**Comment by lexruee** [comment]
@LinxGem33 
It is installed. But I must say that Electron applications are just broken by design :smile:.

**Comment by lestcape** [comment]
@LinxGem33 about of feature (4)... I think will be preferable to have Alacarte, than the menulibre. This is because menulibre is designed to work without any gnome dependency, it will work for xfce, kde... So is difficult to have in the close future something with more customization options.

I'm also working on it... But i have not a big progress. Here is what i have:
https://github.com/lestcape/Classic-Gnome/blob/master/classicGnome%40lestcape/tools/modules/cg_menu.js

This is a submodule of my settings to configured the menus. It's a port of alacarte to gjs. I'm trying to port it, but i don't finished, because we have not an xml parser in gnome shell. So, i port a parser from node.js to gjs also, to then used it in Alacarte: https://github.com/lestcape/Classic-Gnome/tree/92fac148302e130ebd20eba6bebc3bc8790aa901/classicGnome%40lestcape/xml

I hope i can finished this some day...

**Comment by LinxGem33** [comment]
@lexruee 

thanks really appreciate the information regarding feature (4)... i think your right that alacarte is a better option after reading a little bit more about it as its more integrated within the GTK frame work ill look into it and keep you updated regarding this issue.

**Comment by LinxGem33** [comment]
@lexruee 

quick update, I’ve cleaned up the README and updated the information  and Ive also finished the [Reporting-Bug's](https://github.com/LinxGem33/Arc-Menu/wiki/Reporting-Bug's) wiki section and also the [Translation-Guide](https://github.com/LinxGem33/Arc-Menu/wiki/Translation-Guide) wiki section.

**Comment by LinxGem33** [comment]
@lestcape 

hey are you on [wire](https://wire.com/en/download/) chat if so can we add you to our chat group?

**Comment by lestcape** [comment]
@LinxGem33 thanks, but you know, I tell you. i will help when i can, i really have not the time to do more things or be really part of it... I also have some thing similar to your menu here you can see: https://github.com/lestcape/Configurable-Menu/wiki/Windows-7 and it's not work in gnome shell and have for now a low priority to me, so... Really i can not, but really thanks.

**Comment by LinxGem33** [comment]
@lexruee 
left you a message on wire regarding new features.

**Comment by LinxGem33** [comment]
hi @lexruee 

its been a while are you ready for another short round of development for between February and march for the upcoming release of Ubuntu 18.04 LTS let me know if your able to spare a month or so.
kind regards

**Comment by lexruee** [comment]
hi @LinxGem33 
Yeah, I was a little bit occupied with other things :smile:.
I'll have a look at the newly reported issues.

**Comment by LinxGem33** [comment]
@lexruee 
i could tell you have been busy by the amount of projects you now have on your repos 😄 

nice, development will start first week of February until around mid March so not a long time just a short round of development 👍 also can you add the tool tip feature you was working on this release cycle.   

kind regards
